service/header/service.go

```diff
@@ -47,8 +51,8 @@ func (s *Service) Stop(context.Context) error {
 
 // GetByHeight returns the ExtendedHeader at the given height, blocking
 // until header has been processed by the store or context deadline is exceeded.
-func (s *Service) GetByHeight(ctx context.Context, height uint64) (*ExtendedHeader, error) {
-	return s.syncer.store.GetByHeight(ctx, height)
+func (s *Service) GetByHeight(ctx context.Context, height uint64) (*header.ExtendedHeader, error) {
+	return s.syncer.GetByHeight(ctx, height)
```

Comment:
general nit: I find it weird that this method is really on the store but we call it via `syncer`.

Reply:
It is done because service does not have `store` instance. We can change the implementation and request it from network using `Exchange` . @Wondertan wdyt?

Reply:
Why don't we have store inside `header.Service` again @Wondertan ?

Reply:
Well, we should get it from the store directly without syncer.

Reply:
@vgonkivs, the idea about Exchange is good, but not for now. We can make simple logic, if not found in store, requests via Exchange, but we anyway need to check the store for the asked header.